### PR TITLE
Add table aws_elasticsearch_domain. Closes #199

### LINKED ISSUE #199: Add table aws_elasticsearch_domain

### PR DESCRIPTION
**Pull request by anisadas**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
SETUP: tests/aws_elasticsearch_domain []

PRETEST: tests/aws_elasticsearch_domain

TEST: tests/aws_elasticsearch_domain
Running terraform
data.aws_region.alternate: Refreshing state...
data.aws_region.primary: Refreshing state...
data.aws_partition.current: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
data.template_file.resource_aka: Refreshing state...
data.null_data_source.resource: Refreshing state...
aws_elasticsearch_domain.named_test_resource: Creating...
aws_elasticsearch_domain.named_test_resource: Still creating... [10s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [20s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [30s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [40s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [50s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [1m0s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [1m10s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [1m20s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [1m30s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [1m40s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [1m50s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [2m0s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [2m10s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [2m20s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [2m30s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [2m40s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [2m50s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [3m0s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [3m10s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [3m20s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [3m30s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [3m40s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [3m50s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [4m0s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [4m10s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [4m20s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [4m30s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [4m40s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [4m50s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [5m0s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [5m10s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [5m20s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [5m30s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [5m40s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [5m50s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [6m0s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [6m10s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [6m20s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [6m30s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [6m40s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [6m50s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [7m0s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [7m10s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [7m20s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [7m30s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [7m40s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [7m50s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [8m0s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [8m10s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [8m20s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [8m30s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [8m40s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [8m50s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [9m0s elapsed]
aws_elasticsearch_domain.named_test_resource: Creation complete after 9m1s [id=arn:aws:es:us-east-1:986325076436:domain/turbottest32182]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

account_id = 986325076436
aws_partition = aws
aws_region = us-east-1
resource_aka = arn:aws:es:us-east-1:986325076436:domain/turbottest32182
resource_name = turbottest32182

Running SQL query: test-get-query.sql
[
  {
    "account_id": "986325076436",
    "akas": [
      "arn:aws:es:us-east-1:986325076436:domain/turbottest32182"
    ],
    "domain_name": "turbottest32182",
    "ebs_enabled": true,
    "elasticsearch_version": "5.6",
    "partition": "aws",
    "region": "us-east-1",
    "tags": {
      "name": "turbottest32182"
    },
    "volume_size": 10
  }
]
✔ PASSED

Running SQL query: test-hydrate-query.sql
[
  {
    "account_id": "986325076436",
    "akas": [
      "arn:aws:es:us-east-1:986325076436:domain/turbottest32182"
    ],
    "domain_name": "turbottest32182",
    "ebs_enabled": true,
    "elasticsearch_version": "5.6",
    "partition": "aws",
    "region": "us-east-1",
    "tags": {
      "name": "turbottest32182"
    },
    "volume_size": 10
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "account_id": "986325076436",
    "akas": [
      "arn:aws:es:us-east-1:986325076436:domain/turbottest32182"
    ],
    "domain_name": "turbottest32182",
    "partition": "aws",
    "region": "us-east-1"
  }
]
✔ PASSED

Running SQL query: test-notfound-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "arn:aws:es:us-east-1:986325076436:domain/turbottest32182"
    ],
    "tags": {
      "name": "turbottest32182"
    },
    "title": "turbottest32182"
  }
]
✔ PASSED

POSTTEST: tests/aws_elasticsearch_domain

TEARDOWN: tests/aws_elasticsearch_domain

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
select
  domain_name,
  domain_id,
  arn,
  elasticsearch_version,
  created
from
  aws_elasticsearch_domain;

+-------------+-------------------------+-----------------------------------------------------+-----------------------+---------+
| domain_name | domain_id               | arn                                                 | elasticsearch_version | created |
+-------------+-------------------------+-----------------------------------------------------+-----------------------+---------+
| testdomian  | 986325076436/testdomian | arn:aws:es:us-east-2:986325076436:domain/testdomian | 7.9                   | true    |
+-------------+-------------------------+-----------------------------------------------------+-----------------------+---------+

select
  domain_name,
  domain_id,
  encryption_at_rest_options ->> 'Enabled' as enabled,
  encryption_at_rest_options ->> 'KmsKeyId' as kms_key_id
from
  aws_elasticsearch_domain
where
  encryption_at_rest_options ->> 'Enabled' = 'false';
+-------------+--------------------------+---------+------------+
| domain_name | domain_id                | enabled | kms_key_id |
+-------------+--------------------------+---------+------------+
| testdomain  | 986325076436/testdomain  | false   | <null>     |
| turbottest1 | 986325076436/turbottest1 | false   | <null>     |
+-------------+--------------------------+---------+------------+

select
  domain_name,
  domain_id,
  ebs_options ->> 'VolumeSize' as volume_size,
  ebs_options ->> 'VolumeType' as volume_type,
  ebs_options ->> 'EBSEnabled' as ebs_enabled
from
  aws_elasticsearch_domain
where
  ebs_options ->> 'EBSEnabled' = 'true';

+-------------+-------------------------+-------------+-------------+-------------+
| domain_name | domain_id               | volume_size | volume_type | ebs_enabled |
+-------------+-------------------------+-------------+-------------+-------------+
| turbottest  | 986325076436/turbottest | 10          | gp2         | true        |
| testdomian  | 986325076436/testdomian | 10          | gp2         | true        |
+-------------+-------------------------+-------------+-------------+-------------+

select
  domain_name,
  vpc_options ->> 'AvailabilityZones' as availability_zones,
  vpc_options ->> 'SecurityGroupIds' as security_group_ids,
  vpc_options ->> 'SubnetIds' as subnet_ids,
  vpc_options ->> 'VPCId' as vpc_id
from
  aws_elasticsearch_domain
where
  vpc_options ->> 'AvailabilityZones' is not null;

+-------------+------------------------------+--------------------------+----------------------------------------------------------+-----------------------+
| domain_name | availability_zones           | security_group_ids       | subnet_ids                                               | vpc_id                |
+-------------+------------------------------+--------------------------+----------------------------------------------------------+-----------------------+
| testdomian  | ["us-east-2c", "us-east-2a"] | ["sg-024b68b761bd8c164"] | ["subnet-08c8d7312f94703dc", "subnet-0afb90d0b678cf562"] | vpc-032cfaba204ecfa3e |
+-------------+------------------------------+--------------------------+----------------------------------------------------------+-----------------------+

select
  domain_name,
  domain_id,
  elasticsearch_cluster_config ->> 'InstanceType' as instance_type,
  elasticsearch_cluster_config ->> 'InstanceCount' as instance_count
from
  aws_elasticsearch_domain;

+-------------+--------------------------+------------------------+----------------+
| domain_name | domain_id                | instance_type          | instance_count |
+-------------+--------------------------+------------------------+----------------+
| turbottest1 | 986325076436/turbottest1 | t3.small.elasticsearch | 1              |
| testdomain  | 986325076436/testdomain  | t2.small.elasticsearch | 1              |
+-------------+--------------------------+------------------------+----------------+



select
  domain_name,
  p as principal,
  a as action,
  s ->> 'Effect' as effect
from
  aws_elasticsearch_domain,
  jsonb_array_elements(policy_std -> 'Statement') as s,
  jsonb_array_elements_text(s -> 'Principal' -> 'AWS') as p,
  jsonb_array_elements_text(s -> 'Action') as a
where
  p = '*'
  and s ->> 'Effect' = 'Allow';

+-------------+-----------+--------+--------+
| domain_name | principal | action | effect |
+-------------+-----------+--------+--------+
| testdomian  | *         | es:*   | Allow  |
+-------------+-----------+--------+--------+


```
</details>

### CODE REVIEW

File: aws-test/tests/aws_elasticsearch_domain/variables.tf [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  template = "arn:$${partition}:es:$${region}:$${account_id}:domain/$${resource_name}"
+  vars = {
+    resource_name = var.resource_name
+    partition = data.aws_partition.current.partition
+    account_id = data.aws_caller_identity.current.account_id
+    region = data.aws_region.primary.name
+    alternate_region = data.aws_region.alternate.name
+  }
+}
+
```
Comment (Paulami30) [inline aws-test/tests/aws_elasticsearch_domain/variables.tf:78]: terraform returns the arn, you do not need to create the terraform plan

File: aws/table_ aws_elasticsearch_domain.go [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+				Hydrate:     getAwsElasticsearchDomain,
+				Transform:   transform.FromField("VPCOptions.SubnetIds"),
+			},
+			{
+				Name:        "vpc_id",
+				Description: "The VPC Id for the Elasticsearch domain.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getAwsElasticsearchDomain,
+				Transform:   transform.FromField("VPCOptions.VPCId"),
+			},
```
Comment (Paulami30) [inline aws/table_ aws_elasticsearch_domain.go:148]: why did you expand the `VPCOptions` and `EBSOptions` object

File: aws/table_ aws_elasticsearch_domain.go [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("DomainName"),
+			},
+			{
+				Name:        "tags",
+				Description: resourceInterfaceDescription("tags"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     listAwsElasticsearchDomainTags,
+				Transform:   transform.FromField("TagList").Transform(getAwsElasticsearchDomaintagListToTurbotTags),
+			},
```
Comment (Paulami30) [inline aws/table_ aws_elasticsearch_domain.go:231]: arrange the columns as title, tags, akas

File: docs/tables/aws_elasticsearch_domain.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  volume_type,
+  ebs_enabled
+from
+  aws_elasticsearch_domain
+where
+  ebs_enabled = true;
+```
+
+
+### Get details of vpc associated with domain
```
Comment (Paulami30) [inline docs/tables/aws_elasticsearch_domain.md:53]:
```suggestion
### Get network details associated with domain
```

File: docs/tables/aws_elasticsearch_domain.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  encryption_at_rest_options ->> 'Enabled' as enabled,
+  encryption_at_rest_options ->> 'KmsKeyId' as kms_key_id
+from
+  aws_elasticsearch_domain
+where
+  encryption_at_rest_options ->> 'Enabled' = 'false';
+```
+
+
+### Get details of ebs associated with each domain
```
Comment (Paulami30) [inline docs/tables/aws_elasticsearch_domain.md:37]:
```suggestion
### List domains with unencrypted ebs associated with it
```

change the query accordingly

File: aws/table_ aws_elasticsearch_domain.go [hunk 6]
```diff
@@ -0,0 +1,360 @@
...
+	
+	// Get call
+	op, err := svc.ListTags(params)
+	if err != nil {
+		return nil, err
+	}
+
+	return op, nil
+}
+
```
Comment (Paulami30) [inline aws/table_ aws_elasticsearch_domain.go:341]:
```suggestion

//// TRANSFORM FUNCTION
```

**Comment by anisadas** [comment]
<details><summary>Logs after changing in terraform plan</summary>
<p>

```
SETUP: tests/aws_elasticsearch_domain []

PRETEST: tests/aws_elasticsearch_domain

TEST: tests/aws_elasticsearch_domain
Running terraform
data.aws_partition.current: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
data.aws_region.primary: Refreshing state...
data.aws_region.alternate: Refreshing state...
data.null_data_source.resource: Refreshing state...
aws_elasticsearch_domain.named_test_resource: Creating...
aws_elasticsearch_domain.named_test_resource: Still creating... [10s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [20s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [30s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [40s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [50s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [1m0s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [1m10s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [1m20s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [1m30s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [1m40s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [1m50s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [2m0s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [2m10s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [2m20s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [2m30s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [2m40s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [2m50s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [3m1s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [3m11s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [3m21s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [3m31s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [3m41s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [3m51s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [4m1s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [4m11s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [4m21s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [4m31s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [4m41s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [4m51s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [5m1s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [5m11s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [5m21s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [5m31s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [5m41s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [5m51s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [6m1s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [6m11s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [6m21s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [6m31s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [6m41s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [6m51s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [7m1s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [7m11s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [7m21s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [7m31s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [7m41s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [7m51s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [8m1s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [8m11s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [8m21s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [8m31s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [8m41s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [8m51s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [9m1s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [9m11s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [9m21s elapsed]
aws_elasticsearch_domain.named_test_resource: Creation complete after 9m24s [id=arn:aws:es:us-east-1:986325076436:domain/turbottest12463]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

account_id = 986325076436
aws_partition = aws
aws_region = us-east-1
resource_aka = arn:aws:es:us-east-1:986325076436:domain/turbottest12463
resource_name = turbottest12463

Running SQL query: test-get-query.sql
[
  {
    "account_id": "986325076436",
    "akas": [
      "arn:aws:es:us-east-1:986325076436:domain/turbottest12463"
    ],
    "domain_name": "turbottest12463",
    "ebs_enabled": true,
    "elasticsearch_version": "5.6",
    "partition": "aws",
    "region": "us-east-1",
    "tags": {
      "name": "turbottest12463"
    },
    "volume_size": 10
  }
]
✔ PASSED

Running SQL query: test-hydrate-query.sql
[
  {
    "account_id": "986325076436",
    "akas": [
      "arn:aws:es:us-east-1:986325076436:domain/turbottest12463"
    ],
    "domain_name": "turbottest12463",
    "ebs_enabled": true,
    "elasticsearch_version": "5.6",
    "partition": "aws",
    "region": "us-east-1",
    "tags": {
      "name": "turbottest12463"
    },
    "volume_size": 10
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "account_id": "986325076436",
    "akas": [
      "arn:aws:es:us-east-1:986325076436:domain/turbottest12463"
    ],
    "domain_name": "turbottest12463",
    "partition": "aws",
    "region": "us-east-1"
  }
]
✔ PASSED

Running SQL query: test-notfound-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "arn:aws:es:us-east-1:986325076436:domain/turbottest12463"
    ],
    "tags": {
      "name": "turbottest12463"
    },
    "title": "turbottest12463"
  }
]
✔ PASSED

POSTTEST: tests/aws_elasticsearch_domain

TEARDOWN: tests/aws_elasticsearch_domain

SUMMARY:

1/1 passed.
```

</p>
</details>

**Comment by anisadas** [comment]
<details><summary>Logs</summary>
<p>

```
SETUP: tests/aws_elasticsearch_domain []

PRETEST: tests/aws_elasticsearch_domain

TEST: tests/aws_elasticsearch_domain
Running terraform
data.aws_region.alternate: Refreshing state...
data.aws_partition.current: Refreshing state...
data.aws_region.primary: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
data.null_data_source.resource: Refreshing state...
aws_elasticsearch_domain.named_test_resource: Creating...
aws_elasticsearch_domain.named_test_resource: Still creating... [10s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [20s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [30s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [40s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [50s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [1m0s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [1m10s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [1m20s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [1m30s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [1m40s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [1m50s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [2m0s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [2m10s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [2m20s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [2m30s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [2m40s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [2m50s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [3m0s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [3m10s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [3m20s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [3m30s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [3m40s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [3m50s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [4m0s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [4m10s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [4m20s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [4m30s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [4m40s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [4m50s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [5m0s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [5m10s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [5m20s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [5m30s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [5m40s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [5m50s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [6m0s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [6m10s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [6m20s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [6m30s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [6m40s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [6m50s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [7m0s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [7m10s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [7m20s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [7m30s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [7m40s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [7m51s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [8m1s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [8m11s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [8m21s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [8m31s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [8m41s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [8m51s elapsed]
aws_elasticsearch_domain.named_test_resource: Still creating... [9m1s elapsed]
aws_elasticsearch_domain.named_test_resource: Creation complete after 9m9s [id=arn:aws:es:us-east-1:986325076436:domain/turbottest67566]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

account_id = 986325076436
aws_partition = aws
aws_region = us-east-1
resource_aka = arn:aws:es:us-east-1:986325076436:domain/turbottest67566
resource_name = turbottest67566

Running SQL query: test-get-query.sql
[
  {
    "account_id": "986325076436",
    "akas": [
      "arn:aws:es:us-east-1:986325076436:domain/turbottest67566"
    ],
    "domain_name": "turbottest67566",
    "ebs_options": {
      "EBSEnabled": true,
      "Iops": null,
      "VolumeSize": 10,
      "VolumeType": "gp2"
    },
    "elasticsearch_version": "5.6",
    "partition": "aws",
    "region": "us-east-1",
    "snapshot_options": {
      "AutomatedSnapshotStartHour": 23
    },
    "tags": {
      "name": "turbottest67566"
    }
  }
]
✔ PASSED

Running SQL query: test-hydrate-query.sql
[
  {
    "account_id": "986325076436",
    "akas": [
      "arn:aws:es:us-east-1:986325076436:domain/turbottest67566"
    ],
    "domain_name": "turbottest67566",
    "ebs_options": {
      "EBSEnabled": true,
      "Iops": null,
      "VolumeSize": 10,
      "VolumeType": "gp2"
    },
    "elasticsearch_version": "5.6",
    "partition": "aws",
    "region": "us-east-1",
    "snapshot_options": {
      "AutomatedSnapshotStartHour": 23
    },
    "tags": {
      "name": "turbottest67566"
    }
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "account_id": "986325076436",
    "akas": [
      "arn:aws:es:us-east-1:986325076436:domain/turbottest67566"
    ],
    "domain_name": "turbottest67566",
    "partition": "aws",
    "region": "us-east-1"
  }
]
✔ PASSED

Running SQL query: test-notfound-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "arn:aws:es:us-east-1:986325076436:domain/turbottest67566"
    ],
    "tags": {
      "name": "turbottest67566"
    },
    "title": "turbottest67566"
  }
]
✔ PASSED

POSTTEST: tests/aws_elasticsearch_domain

TEARDOWN: tests/aws_elasticsearch_domain

SUMMARY:

1/1 passed.
```
</p>
</details>

File: aws/table_ aws_elasticsearch_domain.go [hunk 7]
```diff
@@ -0,0 +1,333 @@
...
+				Name:        "arn",
+				Description: "The Amazon Resource Name (ARN) of the domain.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getAwsElasticsearchDomain,
+				Transform:   transform.FromField("ARN"),
+			},
+			{
+				Name:        "access_policies",
+				Description: "IAM access policy attach with domain.",
+				Type:        proto.ColumnType_STRING,
```
Comment (Paulami30) [inline aws/table_ aws_elasticsearch_domain.go:54]:
```suggestion
				Type:        proto.ColumnType_JSON,
```

File: aws/table_ aws_elasticsearch_domain.go [hunk 8]
```diff
@@ -0,0 +1,333 @@
...
+				Hydrate:     getAwsElasticsearchDomain,
+			},
+			{
+				Name:        "policy_std",
+				Description: "Contains the policy in a canonical form for easier searching.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getAwsElasticsearchDomain,
+				Transform:   transform.FromField("AccessPolicies").Transform(unescape).Transform(policyToCanonical),
```
Comment (Paulami30) [inline aws/table_ aws_elasticsearch_domain.go:62]: All json columns at the end

File: docs/tables/aws_elasticsearch_domain.md [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+  jsonb_array_elements(policy_std -> 'Statement') as s,
+  jsonb_array_elements_text(s -> 'Principal' -> 'AWS') as p,
+  jsonb_array_elements_text(s -> 'Action') as a
+where
+  p = '*'
+  and s ->> 'Effect' = 'Allow';
+```
+
+
+### List of domains which are plan for deletion
```
Comment (Paulami30) [inline docs/tables/aws_elasticsearch_domain.md:100]:
```suggestion
### List of domains which are planned for deletion
```

File: docs/tables/aws_elasticsearch_domain.md [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+```sql
+select
+  domain_name,
+  domain_id,
+  deleted
+from
+  aws_elasticsearch_domain
+where
+  deleted is not false;
+```
```
Comment (Paulami30) [inline docs/tables/aws_elasticsearch_domain.md:111]: I this query correct?

File: docs/tables/aws_elasticsearch_domain.md [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+
+```sql
+select
+  domain_name,
+  domain_id,
+  deleted
+from
+  aws_elasticsearch_domain
+where
+  deleted is not false;
```
Comment (Paulami30) [inline docs/tables/aws_elasticsearch_domain.md:110]:
```suggestion
  not deleted;
```

File: docs/tables/aws_elasticsearch_domain.md [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+  domain_name,
+  domain_id,
+  elasticsearch_cluster_config ->> 'InstanceType' as instance_type,
+  elasticsearch_cluster_config ->> 'InstanceCount' as instance_count
+from
+  aws_elasticsearch_domain;
+```
+
+
+### List of domains policy statements that grant anonymous access
```
Comment (Paulami30) [inline docs/tables/aws_elasticsearch_domain.md:81]:
```suggestion
### List domains that grant anonymous access
```

**Review by Paulami30 — APPROVED** [review]
LGTM

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**